org-code-javabuilder/protocol/src/main/java/org/code/protocol/JavabuilderThrowableMessageUtils.java

```diff
@@ -35,4 +38,24 @@ public static String getLoggingString(Throwable throwable) {
     throwable.printStackTrace(printWriter);
     return stringWriter.toString();
   }
+
+  /**
+   * @return A pretty shortened version of the exception and stack trace for sharing with end users.
+   *     Private, as this is meant for use as a helper in the context of a client-facing message.
+   */
+  private static String getUserFacingLoggingString(Throwable cause) {
+    if (cause.getClass() == InvocationTargetException.class) {
+      cause = cause.getCause();
+    }
+    StackTraceElement[] stackTrace = cause.getStackTrace();
+    String loggingString = "Exception type: " + cause.toString() + "\n";
```

Comment:
I don't recall what our expectation is around translation in error messages at the moment, I remembered this PR:

https://github.com/code-dot-org/javabuilder/pull/182/files

Do we do translation as a follow-up?

Reply:
Ideally we'd like everything to be translatable. We do have the general issue that a stack trace has some English words in it (such as 'at'), but if we can make a translatable version of the "Exception type: " message that would be helpful. Now that we have the fallback message option I think we can add both this message as the "fallback" and also have a key with parameters for things like the exception name and stack trace, then follow-up with the code-dot-org side change to add the new translatable string.

Reply:
Got it, will update!